src/map/lua/luautils.cpp

```diff
@@ -1517,27 +1517,61 @@ namespace luautils
         lua_pushnil(LuaHandle);
         lua_setglobal(LuaHandle, "onEventUpdate");
 
+        bool luaFileFound = false;
```

Comment:
this kind of turned into a giant shitshow - maybe you can break out the repeated logic into a smaller function?
I can handle that if you don't know how to do it